src/intrinsics/ecma262/Object.js

```diff
@@ -140,7 +143,13 @@ export default function(realm: Realm): NativeFunctionValue {
     }
 
     // 5. Return to.
-    if (to_must_be_partial) to.makePartial();
+    if (to_must_be_partial) {
+      to.makePartial();
+      // if all sources are simple, then the target object must be too
+      if (simple_sources_count === sources.length) {
```

Comment:
I don’t think we know this unless the target had no properties before the call.

Reply:
@gaearon Good point, change made.